source/extensions/filters/http/grpc_field_extraction/filter.cc

```diff
@@ -0,0 +1,229 @@
+#include "source/extensions/filters/http/grpc_field_extraction/filter.h"
+
+#include <memory>
+#include <string>
+
+#include "envoy/http/filter.h"
+#include "envoy/http/header_map.h"
+#include "envoy/registry/registry.h"
+#include "envoy/stream_info/filter_state.h"
+
+#include "source/common/buffer/zero_copy_input_stream_impl.h"
+#include "source/common/common/assert.h"
+#include "source/common/common/base64.h"
+#include "source/common/common/empty_string.h"
+#include "source/common/common/logger.h"
+#include "source/common/grpc/common.h"
+#include "source/common/grpc/status.h"
+#include "source/common/http/codes.h"
+#include "source/common/http/header_map_impl.h"
+#include "source/common/http/header_utility.h"
+#include "source/common/http/headers.h"
+#include "source/common/http/utility.h"
+#include "source/common/protobuf/protobuf.h"
+#include "source/extensions/filters/http/grpc_field_extraction/extractor.h"
+
+#include "absl/strings/escaping.h"
+#include "absl/strings/str_format.h"
+#include "proto_field_extraction/message_data/cord_message_data.h"
+#include "proto_field_extraction/message_data/message_data.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace GrpcFieldExtraction {
+namespace {
+using ::envoy::extensions::filters::http::grpc_field_extraction::v3::FieldExtractions;
+using ::envoy::extensions::filters::http::grpc_field_extraction::v3::GrpcFieldExtractionConfig;
+using ::Envoy::Grpc::Status;
+using ::Envoy::Grpc::Utility;
+
+// The filter prefixes.
+const char kRcDetailFilterGrpcFieldExtraction[] = "grpc_field_extraction";
+
+const char kRcDetailErrorRequestBufferConversion[] = "REQUEST_BUFFER_CONVERSION_FAIL";
+
+const char kRcDetailErrorTypeBadRequest[] = "BAD_REQUEST";
+
+const char kRcDetailErrorRequestFieldExtractionFailed[] = "REQUEST_FIELD_EXTRACTION_FAILED";
+
+const char kRcDetailErrorRequestOutOfData[] = "REQUEST_OUT_OF_DATA";
+
+const char kGrpcFieldExtractionDynamicMetadata[] = "envoy.filters.http.grpc_field_extraction";
+
+std::string generateRcDetails(absl::string_view filter_name, absl::string_view error_type,
+                              absl::string_view error_detail) {
+  return absl::StrCat(filter_name, "_", error_type, "{", error_detail, "}");
+}
+
+// Turns a '/package.Service/method' to 'package.Service.method' which is
```

Comment:
`method` is actually capitalized as `Method` in gRPC path. Can you confirm proto path also expects titlecase and update this comment?

Reply:
This is a recommended convention but lowercase service/method also work. Just changed to /package.service/method.

http://shortn/_L1wy61Bdaz